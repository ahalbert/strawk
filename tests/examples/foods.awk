# If a regex is matched, the full value is put in $0 and each capture group in $1, $2...
/Name: (.*)\nFavorite Food: (.*)\n?/ {
   if ($2 in foods) {
     foods[$2]++
   } else {
     foods[$2] = 1
   }
}

END {
   count, mostpopular = 0, ""
   for (food in foods) {
     if foods[food] > count {
       count = foods[food]
       mostpopular = food
     }
   }
   print "The most popular food is:", mostpopular
}
