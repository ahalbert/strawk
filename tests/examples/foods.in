Name: Bob
Favorite Food: Apple

Name: Alice
Favorite Food: Banana

Name: Pam
Favorite Food: Orange

Name: George
Favorite Food: Banana
