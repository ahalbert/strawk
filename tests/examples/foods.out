The most popular food is: Banana
